languages/ruby/ruby_on_rails.md

```diff
@@ -0,0 +1,124 @@
+# Ruby on Rails Developer Guide
+
+# Overview
+This document is a living doc shared by all developers at Lab Zero.  We as developers at Lab Zero believe that following these principles produces higher quality software.  It’s a pact that we share.  Please read and add to it as we find things on our projects that we’d like to make standard for our engineering practice.
+
+# Like Camping
+When camping there is a rule to leave your campsite cleaner than it was when you got there.  The same is true when writing code.  Each commit should be refactoring the minutia as you go such that each commit leaves the code base cleaner than when you started.
+
+# Documentation
+There are many things that we will need to communicate and share with others developers during and after we’re done.  Please document the basics that help folks know how to install, configure and deploy the app.  We are committed to updating these documents when things change to keep them useful.  Given that’s a hard task, please use documentation wisely.  Too much and it’s unusable and all out of date.  If a dragon hides in the system, write about it.  There are three places that we’ll likely want to send folks.  The app’s README, the repo’s Github wiki, and inline code comments. 
+
+## README’s
+Please delete the Rails boilerplate one when starting a new app and start plumbing it with the what’s where basics.  This page is what’s shown on the Github homepage for the project and is a good launching point to other pages on the wiki. Use Markdown when appropriate to improve formatting.
+
+## Github Wiki
+Anything beyond the uber basics should be documented on the Github wiki pages.  Some examples of this might be “Deployment Anatomy”, “Search Architecture”, “Importing+Refreshing Application Content” etc…
+
+## Code Comments
+Generally these are to be avoided only because they tend to be out of date if used too heavily.  You should favor using descriptive variable and method names.  Rather, please use inline code comments when:
+The following lines of code might use some syntax-fu (aka anything sufficiently clever)
+You’re working around a bug in a 3rd party library/api/model that we can’t control. i.e. When we have to write something that seems bad in order to use an external/immutable dependency.
+Magic numbers, explain your units and why the number exists.
+Class documentation, consider documenting the controller or model to give the big picture so that folks can find their bearing.
+In your Gemfile and you’re locking a gem down to a specific version, please document why we must use that version.  This makes it easier to know when it is safe/reasonable to upgrade the gem.
+ 
+# Tests
+These should be the definition of what the app should does, especially with BDD with Rspec and Cucumber… 
+
+# Code Style
+For sensible suggestions, see https://github.com/bbatsov/ruby-style-guide and https://github.com/bbatsov/rails-style-guide 
+
```

Comment:
It would be nice to link to your new ruby-style-quality-rules.md somewhere in this section.